File: src/java/EntityFX/src/EntityFX/Core/Dhrystone/DhrystoneBenchmark.java
package EntityFX.Core.Dhrystone;

import java.io.FileNotFoundException;
import java.io.IOException;

import EntityFX.Core.Writer;
import EntityFX.Core.Generic.BenchResult;
import EntityFX.Core.Generic.BenchmarkBase;

public class DhrystoneBenchmark extends BenchmarkBase<DhrystoneResult> {

    private Dhrystone2 dhrystone = new Dhrystone2(true);

    public DhrystoneBenchmark(Writer writer, boolean printToConsole) throws FileNotFoundException {
        super(writer, printToConsole);
        Ratio = 4;
    }

    @Override
    public DhrystoneResult benchImplementation() throws IOException {
        return dhrystone.Bench(Dhrystone2.LOOPS);
    }

    @Override
    public BenchResult populateResult(BenchResult benchResult, DhrystoneResult dhrystoneResult) {
        benchResult.Result = dhrystoneResult.VaxMips;
        benchResult.Points = dhrystoneResult.VaxMips * Ratio;
        benchResult.Units = "DMIPS";
        benchResult.Output = dhrystoneResult.Output;
        return benchResult;
    }

    @Override
    public void warmup(Double aspect) {
    }
}

File: src/python/entityfx/dhrystone_benchmark.py

from entityfx.benchmark_base import BenchmarkBase
from entityfx.dhrystone2 import Dhrystone2
from entityfx.writer import Writer


class DhrystoneBenchmark(BenchmarkBase):
    
    def __init__(self, writer: Writer, print_to_console : bool=True) -> None:
        super().__init__(writer, print_to_console)
        self.__dhrystone = Dhrystone2()
        self.ratio = 4
    
    def benchImplementation(self):
        return self.__dhrystone.bench()
    
    def populateResult(self, bench_result, dhrystone_result):
        bench_result["Points"] = dhrystone_result["VaxMips"]
        bench_result["Result"] = dhrystone_result["VaxMips"] * self.ratio
        bench_result["Units"] = "DMIPS"
        bench_result["Output"] = dhrystone_result["Output"]
        return bench_result
    
    def warmup(self, aspect : float=.05) -> None:
        pass
